modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -1118,10 +1118,16 @@ class TiffEncoderBufHelper
                                /*map=*/0, /*unmap=*/0 );
     }
 
-    static tmsize_t read( thandle_t /*handle*/, void* /*buffer*/, tmsize_t /*n*/ )
+    static tmsize_t read( thandle_t handle, void* buffer, tmsize_t n )
     {
-        // Not used for encoding.
-        return 0;
+        // Used for imencodemulti() to stores multi-images.
+        TiffEncoderBufHelper *helper = reinterpret_cast<TiffEncoderBufHelper*>(handle);
+        size_t begin = (size_t)helper->m_buf_pos;
+        size_t end = begin + n;
+        CV_CheckGT( helper->m_buf->size(), end , "do not be over-run buffer");
+        memcpy(buffer, &(*helper->m_buf)[begin], n);
```

Comment:
std::memcpy to stay C++

Reply:
I'm very sorry if my understanding is wrong.
I replaced from `memcpy` to `std::memcpy` in grfmt_tiff.cpp and re-indent..